### DISCUSSION TITLE
Follow doesn't follow

### DISCUSSION THREAD

**Discussion by timtuxworth** [stronnag/mwptools#132] [General]
I just installed mwptools on a fresh install of Ubuntu 22.04 and connected my TX16 S running Open TX 2.3.15. I enabled Telemetry forwarding on the transmitter and wmptools can see the telemetry and displays vehicle attitude in the HUD and  the GPS location in the applets, but the map doesn't move/reposition, even though the "follow" option is checked. I tried checking/unchecking the box.

**Comment by stronnag** [comment]
`Follow` means that the craft location is updated with a valid GPS fix
`In View` means the map will be panned to keep `Follow` in view
`use-legacy-centre-on` keeps the vehicle centred and moves the map, which is more CPU intensive (and may induce motion sickness in the observer).

If you still can't set this to work, a dump of the settings (`gsettings list-recursively org.mwptools.planner`) and a recording of the telemetry (`mwp --raw-log`) would help.

**Comment by timtuxworth** [comment]
When map starts up it's centered in Europe somewhere. My location is in Western Canada. I'm just doing this on the bench atm, but the map doesn't reposition to where I am, even though I have both Follow and In View checked. I tried mwp --use-legacy-center-on but there is no such option.
![Screenshot 2023-01-31 at 10 44 42 AM](https://user-images.githubusercontent.com/5266784/215842101-cc56c6f2-31ef-4c16-938a-50b3fef75f8e.png)

**Comment by snaewe** [comment]
It's called 'use-legacy-centre-on' (yes, centre not center).

**Comment by timtuxworth** [comment]
![Screenshot 2023-01-31 at 10 54 50 AM](https://user-images.githubusercontent.com/5266784/215843379-d937eba8-07ae-4951-8d54-7a534985c0c9.png)

**Comment by stronnag** [comment]
Please see the preferences for setting a default location.
`use-legacy-centre-on` is a setting not a option.
It is also spelled in English.

As you're not supplied the requested data, I shall close this.

**Comment by timtuxworth** [comment]
These are the logs 
[mwp_2023-01-31_105342.log](https://github.com/stronnag/mwptools/files/10549401/mwp_2023-01-31_105342.log)

https://www.dropbox.com/s/xip7vicdcjikme3/mwp._dev_ttyACM0.2023-01-31T105323.raw?dl=0

**Comment by timtuxworth** [comment]
Here is a new log after turning on use-legacy-centre-on in the settings 
[mwp._dev_ttyACM0.2023-01-31T110325.raw](https://www.dropbox.com/s/ob0nt6ecpbvs21n/mwp._dev_ttyACM0.2023-01-31T110325.raw?dl=0)
and a stderr.txt file that was generated:
https://www.dropbox.com/s/alw23lasrdpy40t/mwp_stderr_2023-01-31.txt?dl=0

**Comment by stronnag** [comment]
Please also provide the INAV FC CLI output of  `status` and `version`.

**Reply by timtuxworth** [reply]
I'm running ArduPilot with ExpressLRS

**Reply by stronnag** [reply]
> I'm running ArduPilot

Actually, I'd worked that out from `  49.06s: FM: FBWA` from the CRSF raw file; confirmation appreciated.

From mwp's fromt page _"ground` station, mission planner and tools for inav and multiwii-nav"_

Now we've established what's happening, I'm sort of interested in getting mwp to work with AP's CRSF. 

And if I replay your mwp JSON log, see below. Looks like Canada to me. 

![image](https://user-images.githubusercontent.com/158229/215848306-833ba657-b7dd-4149-84ed-996f5372b890.png)

I will create an issue "Support AP flavour of CRSF" which is the real issue.

**Reply by timtuxworth** [reply]
Yup - Edmonton, Aberta, Canada :-)

**Comment by stronnag** [comment]
More questions:

Is there any documentation on the flight modes provided by AP's CRSF ?

If I replay your CRSF log, the map does not zoom to the location. Possibly because it doesn't understand the flight mode.

**Comment by timtuxworth** [comment]
These: https://ardupilot.org/plane/docs/flight-modes.html

**Comment by stronnag** [comment]
Continued in #133
